src/Sylius/Bundle/AdminBundle/Console/Command/ListAdminUsersCommand.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Console\Command;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Component\User\Repository\UserRepositoryInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Helper\TableStyle;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+#[AsCommand(
+    name: 'sylius:admin-user:list',
+    description: 'Displays a list of all admin users along with their details.',
+)]
+final class ListAdminUsersCommand extends Command
+{
+    protected SymfonyStyle $io;
+
+    /** @param UserRepositoryInterface<AdminUserInterface> $adminUserRepository */
+    public function __construct(
+        private readonly UserRepositoryInterface $adminUserRepository,
+        private readonly EntityManagerInterface $adminManager,
+    ) {
+        parent::__construct();
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->io = new SymfonyStyle($input, $output);
+    }
+
+    protected function configure(): void
+    {
+        $this->addOption(
+            'search',
+            's',
+            InputOption::VALUE_OPTIONAL,
+            'Filter users based on a search term. If not provided, all users will be displayed.',
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->io->title('List available admin users');
+
+        $searchTerm = $input->getOption('search');
+        if ($searchTerm === null || $searchTerm === '') {
+            $this->renderUsersTable($output, $this->adminUserRepository->findAll());
+
+            return Command::SUCCESS;
+        }
+
+        $users = $this->searchUsers($searchTerm);
+        if (empty($users)) {
+            $this->io->warning(sprintf('No users found matching "%s"', $searchTerm));
+
+            return Command::SUCCESS;
+        }
```

Comment:
Now, that code was refactored few times, I think logic branching for all and filtered users is not necessary anymore

```suggestion
        $adminUsers = $this->findAdminUsers($input->getOption('search'));

        if (empty($adminUsers)) {
            $this->io->warning('No admin users found.');

            return Command::SUCCESS;
        }
```

and

```php
    /**
     * @return array<AdminUserInterface>
     */
    private function findAdminUsers(?string $criteria): array
    {
        if (empty($criteria)) {
            return $this->adminUserRepository->findAll();
        }

        return $this->adminUserManager->createQueryBuilder()
            ->from(AdminUserInterface::class, 'o')
            ->select('o')
            ->where('LOWER(o.email) LIKE :criteria
                OR LOWER(o.username) LIKE :criteria
                OR LOWER(o.firstName) LIKE :criteria
                OR LOWER(o.lastName) LIKE :criteria')
            ->setParameter('criteria', '%' . mb_strtolower($searchTerm) . '%')
            ->orderBy('o.createdAt', 'DESC')
            ->getQuery()
            ->getResult();
    }
```

Reply:
I'm confused. Why should the filter logic removed?

Reply:
Also in your suggestion you also removed the renderUsersTable method which is needed to render the table..

Reply:
I’m not suggesting removing the filtering logic, but rather refactoring the code. It has been modified several times, and as a result, the overall clarity and structure have been affected.